frontend/src/app/modules/time-series/services/line-chart.service.ts

```diff
@@ -1050,142 +1020,142 @@ export class LineChartService {
+      const label: HTMLTableCellElement = document.createElement('td');
+      label.append(currentPoint.tooltipText);
+
+      const value: HTMLTableCellElement = document.createElement('td');
+      const lineColorDot: HTMLElement = document.createElement('i');
+      lineColorDot.className = 'fas fa-circle';
+      lineColorDot.style.color = d3Select(node).style('stroke');
+      value.append(lineColorDot);
+      if (currentPoint.value !== undefined && currentPoint.value !== null) {
+        value.append(currentPoint.value.toString());
+      }
 
-  private generateTooltipTimestampRow(highlightedDate: Date): string | Node {
-    const label: HTMLTableCellElement = document.createElement('td');
-    label.append('Timestamp');
+      const row: HTMLTableRowElement = document.createElement('tr');
+      if (currentPoint.equals(nearestDotData)) {
+        row.className = 'active';
+      }
+      row.append(label);
+      row.append(value);
+      return row;
+    };
 
-    const date: HTMLTableCellElement = document.createElement('td');
-    date.append(highlightedDate.toLocaleString());
+    return (nearestDot: D3Selection<any, TimeSeriesPoint, null, undefined>, visibleDots: D3Selection<D3BaseType, {}, HTMLElement, any>, highlightedDate: Date): HTMLTableElement => {
+      const nearestDotData = nearestDot.datum() as TimeSeriesPoint;
 
-    const row: HTMLTableRowElement = document.createElement('tr');
-    row.append(label);
-    row.append(date);
-    return row;
-  }
+      const table: HTMLTableElement = document.createElement('table');
+      const tableBody: HTMLTableSectionElement = document.createElement('tbody');
+      table.append(tableBody);
+      tableBody.append(generateTooltipTimestampRow(highlightedDate));
 
-  private generateTooltipDataPointRow(currentPoint: TimeSeriesPoint, node: D3BaseType, nearestDotData: TimeSeriesPoint): string | Node {
-    const label: HTMLTableCellElement = document.createElement('td');
-    label.append(currentPoint.tooltipText);
-
-    const value: HTMLTableCellElement = document.createElement('td');
-    const lineColorDot: HTMLElement = document.createElement('i');
-    lineColorDot.className = 'fas fa-circle';
-    lineColorDot.style.color = d3Select(node).style('stroke');
-    value.append(lineColorDot);
-    if (currentPoint.value !== undefined && currentPoint.value !== null) {
-      value.append(currentPoint.value.toString());
-    }
+      const tempArray = [];
+      visibleDots
+        .each((timeSeriesPoint: TimeSeriesPoint, index: number, nodes: D3BaseType[]) => {
+          tempArray.push({
+            "value": timeSeriesPoint.value,
+            "htmlNode": generateTooltipDataPointRow(timeSeriesPoint, nodes[index], nearestDotData)
+          });
+        });
+      tempArray
+        .sort((a, b) => b.value - a.value)
+        .forEach(elem => table.append(elem.htmlNode));
```

Comment:
Shouldn't the elements be appended to the `tableBody` ?

```html
 .forEach(elem => tableBody.append(elem.htmlNode));
                   ^^^^^^
```

Reply:
Currently the implementation results in two `tbody` elements in the dom:
![image](https://user-images.githubusercontent.com/5690260/72792809-e3bf7800-3c39-11ea-9396-678bccaf8d3b.png)

Reply:
Done.